widgetastic_manageiq/__init__.py

```diff
@@ -4356,3 +4356,145 @@ def fill(self, value):
+            if self.browser.text(el) == vm_name:
+                return item
+        raise ItemNotFound("VM: {} not found".format(vm_name))
+
+    def read(self):
+        return self.all_items
+
+    def fill(self, value):
+        # Fill operation is not applicable as the item is not clickable
+        pass
+
+    def get_message_text(self, vm_name):
+        try:
+            el = self._get_vm_element(vm_name)
+            return self.browser.text(self.browser.element(self.ITEM_MESSAGE_LOCATOR,
+                 parent=el))
+        except NoSuchElementException:
+            # Plan not started yet
+            return None
+
+    def get_clock(self, vm_name):
+        el = self._get_vm_element(vm_name)
+        return self.browser.text(self.browser.element(self.ITEM_TIMER_LOCATOR,
+            parent=el))
+
+    def open_additional_info_popup(self, vm_name):
+        try:
+            el = self._get_vm_element(vm_name)
+            self.browser.element(self.ITEM_ADDITIONAL_INFO_BUTTON_LOCATOR, parent=el).click()
+        except NoSuchElementException:
+            # Plan not started yet
+            return None
+
+    def get_progress_description(self, vm_name):
+        el = self._get_vm_element(vm_name)
+        return self.browser.text(self.browser.element(self.ITEM_PROGRESS_DESCRIPTION_LOCATOR,
+            parent=el))
+
+    def download_log(self, vm_name):
+        try:
+            el = self._get_vm_element(vm_name)
+            self.browser.element(self.ITEM_DOWNLOAD_LOG_BUTTON_LOCATOR, parent=el).click()
+        except NoSuchElementException:
+            # Plan not started yet
+            return None
+
+    def is_errored(self, vm_name):
+        """See if cross icon indicating failure present."""
+        try:
+            el = self._get_vm_element(vm_name)
+            return self.browser.element(self.ITEM_IS_ERRORED_LOCATOR, parent=el).is_displayed()
+        except NoSuchElementException:
+            # This means element not present, so no error
+            return False
+
+    def is_in_progress(self, vm_name):
+        """ Checks if spinner is present indicating progress."""
+        try:
+            el = self._get_vm_element(vm_name)
+            return self.browser.element(self.ITEM_PROGRESS_SPINNER_LOCATOR,
+             parent=el).is_displayed()
+        except NoSuchElementException:
+            # This means element not present, so not in-progress
+            return False
+
+    def is_successful(self, vm_name):
+        """ Checks if check icon is present indicating success."""
+        try:
+            el = self._get_vm_element(vm_name)
+            return self.browser.element(self.ITEM_IS_SUCCESSFUL_LOCATOR,
+             parent=el).is_displayed()
+        except NoSuchElementException:
+            # This means element not present, so plan may not have started
+            return False
+
+
+class MigrationPlanRequestDetailsPaginator(Paginator):
```

Comment:
**REQUIRED**
It's better to put these pagination widgets into cfme/v2v/migrations.py